core/src/main/java/org/apache/iceberg/rest/RESTMetricsReporter.java

```diff
@@ -26,7 +26,11 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-class RESTMetricsReporter implements MetricsReporter {
+/**
+ * A {@link MetricsReporter} implementation that reports the {@link MetricsReport} to a REST
+ * endpoint. It's the default when using the {@link RESTCatalog}.
+ */
+public class RESTMetricsReporter implements MetricsReporter {
```

Comment:
I don't think we should make it public here.
I would rather remove the corresponding javadoc (to avoid to confuse users).

Reply:
I think pointing it to the Java source in Github would be better indeed 👍

Reply:
I've pointed all links in `metrics-reporting.md` to GitHub.

Reply:
I agree, we should just point to the source code in GH rather than making the class itself public